src/analysis/normed_space/compact_operator.lean

```diff
@@ -0,0 +1,375 @@
+  [topological_space M₃] [add_comm_monoid M₁] [add_comm_monoid M₂] [add_comm_monoid M₃]
+  [module R₁ M₁] [module R₂ M₂] [module R₃ M₃]
+
+/-- If a compact operator preserves a closed submodule, its restriction to that submodule is
+compact. -/
+lemma is_compact_operator.restrict_invariant {f : M₁ →ₗ[R₁] M₁} (hf : is_compact_operator f)
+  {V : submodule R₁ M₁} (hV : ∀ v ∈ V, f v ∈ V) (h_closed : is_closed (V : set M₁)):
+  is_compact_operator (f.restrict hV) :=
+(hf.comp_clm V.subtypeL).cod_restrict (set_like.forall.2 hV) h_closed
+
+/-- If a compact operator preserves a complete submodule, its restriction to that submodule is
+compact. -/
+lemma is_compact_operator.restrict_invariant' [separated_space M₂] {f : M₂ →ₗ[R₂] M₂}
+  (hf : is_compact_operator f) {V : submodule R₂ M₂} (hV : ∀ v ∈ V, f v ∈ V)
+  [hcomplete : complete_space V] : is_compact_operator (f.restrict hV) :=
+hf.restrict_invariant hV (complete_space_coe_iff_is_complete.mp hcomplete).is_closed
+
+end restrict_invariant
+
+section continuous
+
+variables {𝕜₁ 𝕜₂ : Type*} [nontrivially_normed_field 𝕜₁] [nontrivially_normed_field 𝕜₂]
+  {σ₁₂ : 𝕜₁ →+* 𝕜₂} [ring_hom_isometric σ₁₂] {M₁ M₂ : Type*} [topological_space M₁]
+  [add_comm_group M₁] [topological_space M₂] [add_comm_group M₂] [module 𝕜₁ M₁] [module 𝕜₂ M₂]
+  [topological_add_group M₁] [has_continuous_const_smul 𝕜₁ M₁]
+  [topological_add_group M₂] [has_continuous_smul 𝕜₂ M₂]
+
+@[continuity] lemma is_compact_operator.continuous {f : M₁ →ₛₗ[σ₁₂] M₂}
+  (hf : is_compact_operator f) : continuous f :=
+begin
+  letI : uniform_space M₂ := topological_add_group.to_uniform_space _,
+  haveI : uniform_add_group M₂ := topological_add_group_is_uniform,
+  refine continuous_of_continuous_at_zero f (λ U hU, _),
+  rw map_zero at hU,
+  rcases hf with ⟨K, hK, hKf⟩,
+  rcases hK.totally_bounded.is_vonN_bounded 𝕜₂ hU with ⟨r, hr, hrU⟩,
+  rcases normed_field.exists_lt_norm 𝕜₁ r with ⟨c, hc⟩,
+  have hcnz : c ≠ 0 := ne_zero_of_norm_ne_zero (hr.trans hc).ne.symm,
+  suffices : (σ₁₂ $ c⁻¹) • K ⊆ U,
+  { refine mem_of_superset _ this,
+    have : is_unit c⁻¹ := hcnz.is_unit.inv,
+    rwa [mem_map, preimage_smul_setₛₗ _ _ _ f this, set_smul_mem_nhds_zero_iff (inv_ne_zero hcnz)],
+    apply_instance },
+  rw [σ₁₂.map_inv, ← subset_set_smul_iff₀ (σ₁₂.map_ne_zero.mpr hcnz)],
+  refine hrU (σ₁₂ c) _,
+  rw ring_hom_isometric.is_iso,
+  exact hc.le
+end
+
+/-- Upgrade a compact `linear_map` to a `continuous_linear_map`. -/
+def continuous_linear_map.mk_of_is_compact_operator {f : M₁ →ₛₗ[σ₁₂] M₂}
+  (hf : is_compact_operator f) : M₁ →SL[σ₁₂] M₂ :=
+⟨f, hf.continuous⟩
+
+end continuous
+
+lemma is_closed_set_of_is_compact_operator {𝕜₁ 𝕜₂ : Type*} [nontrivially_normed_field 𝕜₁]
+  [nontrivially_normed_field 𝕜₂] {σ₁₂ : 𝕜₁ →+* 𝕜₂} [ring_hom_isometric σ₁₂] {M₁ M₂ : Type*}
+  [seminormed_add_comm_group M₁] [normed_add_comm_group M₂] [normed_space 𝕜₁ M₁]
+  [normed_space 𝕜₂ M₂] [complete_space M₂] :
+  is_closed {f : M₁ →SL[σ₁₂] M₂ | is_compact_operator f} :=
```

Comment:
Could you add a corollary about the `topological_closure` of the submodule of compact operators?